books/views/catalog.py

```diff
@@ -36,16 +36,18 @@ def index(request: HttpRequest) -> HttpResponse:
     '''Index page, starting page'''
     # Getting all Tags and creating querystring objects for each to pass to template
     tags_to_render = []
-    books = Book.objects.active_books_ordered_by_date()
+    books = Book.objects.active_books_ordered_by_date(['authors', 'narrations'])
     for tag in Tag.objects.filter(name__in=TAGS_TO_SHOW_ON_MAIN_PAGE):
+        books_by_tag = books.filter(tag=tag.id)
+        count_books_by_tag = len(books_by_tag)
```

Comment:
Won't this require loading all books for a given tag? Or does it optimize it somehow and only runs "count"?

The current (previous) version is that we have two queries:
1. First to load first 6 books.
2. Second to count number of books.

WIth the new approach we have just one query. But it loads ~200 books. 

Though I'm not 100% sure as I haven't looked at the actual queries.

Reply:
Your assumptions are absolutely right, but since DB still small (I guess postgres keeps it whole in memory) , data volume does not affect performance while network hops between backend app and db instance do. That version may looks a bit hacky because it works only for current conditions, but you can share your expectations about DB grow and we can make a decision... trade-off as always.

Btw, I might be wrong to and would be good to test this version in staging environment if you have such. We remember first version was rendering in 400ms, right? Can you check how fast this one will work?

Reply:
Right now the database contains 573 books. I expect around 100 new books a year. I don't know at which point it makes sense to switch to two queries as data becomes too big. You understand that better. If for current 1 query and larger response is better - sounds good. 

I'll deploy it to the staging.